MapleServer2/Managers/FieldManager.cs

```diff
@@ -157,18 +163,26 @@ public IFieldActor<NpcMetadata> RequestMob(int mobId, CoordF coord = default, Co
         return mob;
     }
 
-    public IFieldActor<NpcMetadata> RequestMob(int mobId, IFieldObject<MobSpawn> spawnPoint, CoordF coord = default, CoordF rotation = default,
-        short animation = default)
+    public IFieldActor<NpcMetadata> RequestMob(int mobId, IFieldObject<MobSpawn> spawnPoint)
     {
         Mob mob = WrapMob(mobId);
         mob.OriginSpawn = spawnPoint;
-        mob.Coord = coord;
-        mob.Rotation = rotation;
-        if (animation != default)
+
+        Shape shape = Navigator.AddShape(mob.Value.NpcMetadataCapsule);
+        CoordS? randomPositionAround = Navigator.FindClosestUnobstructedCoordS(shape, spawnPoint.Coord.ToShort(), spawnPoint.Value.SpawnRadius);
```

Comment:
having an implicit constructor between CoordF and CoordS might make usage cleaner.

Logic used by client for conversion:
```
short x = xf + xf >= 0 ? 0.001 : -0.001;
short y = yf + yf >= 0 ? 0.001 : -0.001;
short z = zf + 0.001;
```